airbyte-workers/src/main/java/io/airbyte/workers/temporal/scheduling/activities/ConfigFetchActivityImpl.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2021 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.workers.temporal.scheduling.activities;
+
+import io.airbyte.config.Configs;
+import io.airbyte.config.StandardSync;
+import io.airbyte.config.helpers.ScheduleHelpers;
+import io.airbyte.config.persistence.ConfigNotFoundException;
+import io.airbyte.config.persistence.ConfigRepository;
+import io.airbyte.scheduler.models.Job;
+import io.airbyte.scheduler.persistence.JobPersistence;
+import io.airbyte.validation.json.JsonValidationException;
+import io.airbyte.workers.temporal.exception.RetryableException;
+import java.io.IOException;
+import java.time.Duration;
+import java.util.Optional;
+import java.util.function.Supplier;
+import lombok.AllArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+
+@AllArgsConstructor
+@Slf4j
+public class ConfigFetchActivityImpl implements ConfigFetchActivity {
+
+  private ConfigRepository configRepository;
+  private JobPersistence jobPersistence;
+  private Configs configs;
+  private Supplier<Long> currentSecondsSupplier;
+
+  @Override
+  public ScheduleRetrieverOutput getTimeToWait(final ScheduleRetrieverInput input) {
+    try {
+      final Optional<Job> previousJobOptional = jobPersistence.getLastReplicationJob(input.getConnectionId());
+      final StandardSync standardSync = configRepository.getStandardSync(input.getConnectionId());
+
+      if (previousJobOptional.isEmpty() && standardSync.getSchedule() != null) {
+        // Non-manual syncs don't wait for their first run
+        return new ScheduleRetrieverOutput(Duration.ZERO);
+      }
+      if (standardSync.getSchedule() == null) {
+        // Manual syncs wait for their first run
+        return new ScheduleRetrieverOutput(Duration.ofDays(100 * 365));
```

Comment:
Is this saying that manual syncs wait 100 years for their first run? Is that true, or is it just a placeholder to say that they will never run automatically, since they need to be manually triggered?

Reply:
Also it seems like we could move this if statement and the `standardSync` query to the top of the method, so that we avoid querying the jobs database in this case

Reply:
> 

It is a placeholder to say that a manual run won't be automatically trigger after the creation is created.

Reply:
> Also it seems like we could move this if statement and the standardSync query to the top of the method, so that we avoid querying the jobs database in this case

Thanks for the catch, Done.

Reply:
> It is a placeholder to say that a manual run won't be automatically trigger after the creation is created.

In that case I think it may be slightly better to have this return something like `null` or `-1` to indicate that it will not be automatically triggered (and then add logic to handle that value correctly in that case). Returning 100 years feels strange to me